sstables/column_translation.hh

```diff
@@ -64,6 +64,7 @@ private:
         std::vector<column_info> regular_schema_columns_from_sstable;
         std::vector<column_info> static_schema_columns_from_sstable;
         column_values_fixed_lengths clustering_column_value_fix_lengths;
+        bool _empty = false;
```

Comment:
Wouldn't using `optional<column_translation>` be a bit more elegant than building in "optionality" into `column_translation`?

Reply:
I wanted to reduce overhead on the hot path to minimum.  optional<> would make it slightly heavier to copy.

Reply:
Do we copy column_translation in the hot path? I think we share it instead.

Reply:
We share it, column_translation is a shared pointer. So Kamil is suggesting passing around optional<lw_shared_ptr<...>> instead of lw_shared_ptr<...>

Reply:
So why bool _empty?

Reply:
Existing code assumes it's always != nullptr, because column_translation also acts as a cache:

```
    lw_shared_ptr<const state> _state = make_lw_shared<const state>();

public:
    column_translation get_for_schema(
            const schema& s, const serialization_header& header, const sstable_enabled_features& features) {
        if (s.version() != _state->schema_uuid) {
            _state = make_lw_shared<const state>(s, header, features);
        }
        return *this;
    }
```

I could refactor it. But adding "empty" fits into current model, we cache the fact that the mapping for this schema format doesn't exist. If we reused nullptr, it could mean both "mapping not computed" or "format is < mc". This adds an extra conditional to get_column_mapping().

Reply:
It's not used with ka/kl. Some paths use the new empty() to determine that, some paths (already) check sstable version and choose a different impl.

Reply:
ok